clusterloader2/pkg/config/codec.go

```diff
@@ -35,12 +35,11 @@ var (
 )
 
 // convertToConfig converts array of bytes into test config.
-func convertToConfig(raw []byte) (*api.Config, error) {
-	var config api.Config
-	if err := yaml.NewYAMLOrJSONDecoder(bytes.NewBuffer(raw), 4096).Decode(&config); err != nil {
-		return nil, fmt.Errorf("decoding failed: %v", err)
+func convertToConfig(config *api.Config, raw []byte) error {
```

Comment:
Why did you change this?

Reply:
@wojtek-t  Because the default values are set before parsing the testcong yaml file. So  I'll need to define testConfig in test/test.go and pass the pointer to TemplateToConfig in config/template_provider.go. The current version is to define testConfig in config/codec.go and return the pointer to it.

Reply:
Changed it to parse confit first and then set defaults for unset Parma.